local-cli/bundle/buildBundle.js

```diff
@@ -25,7 +25,12 @@ function buildBundle(args, config, output = outputBundle, packagerInstance) {
 
     // This is used by a bazillion of npm modules we don't control so we don't
     // have other choice than defining it as an env variable here.
-    process.env.NODE_ENV = args.dev ? 'development' : 'production';
+    if (!process.env.NODE_ENV) {
+      // If you're inlining environment variables, you can use babel to remove
+      // this line:
+      // https://www.npmjs.com/package/babel-remove-process-env-assignment
```

Comment:
This line will still cause a syntax error when using https://www.npmjs.com/package/babel-plugin-transform-inline-environment-variables , so you can use the specified babel plugin if you're like me and can't stand the stench of mutating global state.